src/components/ModalPortal.js

```diff
@@ -299,6 +300,15 @@ export default class ModalPortal extends Component {
     this.shouldClose = null;
   };
 
+  handleOverlayRightClick = (event) => {
+    if (this.shouldClose === null) this.shouldClose = true;
+    else if (!this.shouldClose) this.shouldClose = null;
+    if (this.shouldClose) {
+      event.preventDefault();
+      this.props.onOverlayRightClick(event);
+    }
+  };
```

Comment:
hmmm...:thinking: I don't think we can initiate the closing process here. It must happend on the user side. So, the only thing we can do here is to forward the callback the user defined for `onOverlayRightClick`.

Reply:
If someone want to just use the right click for other purpose, this will conflict.

Reply:
@diasbruno Kindly check my latest commit. I think now it's okay.